layers.go

```diff
@@ -970,6 +979,66 @@ func (r *layerStore) SetNames(id string, names []string) error {
 	return ErrLayerUnknown
 }
 
+func (r *layerStore) datadir(id string) string {
+	return filepath.Join(r.layerdir, id)
+}
+
+func (r *layerStore) datapath(id, key string) string {
+	return filepath.Join(r.datadir(id), makeBigDataBaseName(key))
+}
+
+func (r *layerStore) BigData(id, key string) ([]byte, error) {
+	if key == "" {
+		return nil, errors.Wrapf(ErrInvalidBigDataName, "can't retrieve layer big data value for empty name")
+	}
+	layer, ok := r.lookup(id)
+	if !ok {
+		return nil, errors.Wrapf(ErrLayerUnknown, "error locating layer with ID %q", id)
+	}
+	return ioutil.ReadFile(r.datapath(layer.ID, key))
+}
+
+func (r *layerStore) SetBigData(id, key string, data []byte) error {
```

Comment:
Same here, we could turn `data` into an `io.Reader`?

Reply:
I am fine with it, just it will be different than all the others `BigData()/SetBigData()` functions we already have for containers and images

Reply:
changed to use `io.Reader and io.ReadCloser`